typing/typecore.ml

```diff
@@ -1103,24 +1198,25 @@ and type_pat_aux ~exception_allowed ~constrs ~labels ~no_existentials ~mode
       | _ -> assert false
       end
   | Ppat_alias(sq, name) ->
-      assert (constrs = None);
-      type_pat sq expected_ty (fun q ->
+      assert (data.constrs = None);
+      (* LATER: "alias" could be allowed in computation patterns *)
+      type_pat Value data sq expected_ty (fun q ->
         begin_def ();
         let ty_var = build_as_type !env q in
         end_def ();
         generalize ty_var;
         let id =
           enter_variable ~is_as_variable:true loc name ty_var sp.ppat_attributes
         in
-        rp k {
+        k @@ rp @@ pure category {
           pat_desc = Tpat_alias(q, id, name);
           pat_loc = loc; pat_extra=[];
           pat_type = q.pat_type;
           pat_attributes = sp.ppat_attributes;
           pat_env = !env })
   | Ppat_constant cst ->
       let cst = constant_or_raise !env loc cst in
-      rup k {
+      k @@ rp @@ pure category @@ unif {
```

Comment:
Very impressive (and successful) attempt at obfuscation.

Reply:
The function `rup k` is equivalent to `k ++ rp ++ unif` (I'm writing `++` for function composition in the usual mathematical order), I need to add `pure category` in the middle, only at some of the callsites. I can understand not being used to this style of chained applications, but I wouldn't call it obfuscation.